pennylane/circuit_graph.py

```diff
@@ -318,25 +282,23 @@ def descendants(self, ops):
         Returns:
             list[Operator]: descendants of the given operators
         """
-        # rx.descendants() returns node indices instead of node-values
-        all_indices = set().union(*(rx.descendants(self._graph, self._indices[id(o)]) for o in ops))
-        double_op_indices = set(self._indices[id(o)] for o in ops)
-        ancestor_indices = all_indices - double_op_indices
-
-        return list(self._graph.get_node_data(n) for n in ancestor_indices)
-
-    def _in_topological_order(self, ops):
-        """Sorts a set of operators in the circuit in a topological order.
-
-        Args:
-            ops (Iterable[Operator]): set of operators in the circuit
-
-        Returns:
-            Iterable[Operator]: same set of operators, topologically ordered
-        """
-        G = self._graph.subgraph(list(self._indices[id(o)] for o in ops))
-        indexes = rx.topological_sort(G)
-        return list(G[x] for x in indexes)
+        if isinstance(ops, (Operator, MeasurementProcess)):
+            raise ValueError(
+                "CircuitGraph.descendants accepts an iterable of"
+                " operators and measurements, not operators and measurements themselves."
+            )
+        if any(len(self._inds_for_objs[WrappedObj(op)]) > 1 for op in ops):
+            raise ValueError(
+                "cannot calculate decendents for an operator that occurs multiple times."
+            )
```

Comment:
Similar to `ancestors`, could this be all descendants of the first occurrence minus all the later occurrences?